app/scripts/modules/core/src/api/ApiService.spec.ts

```diff
@@ -20,6 +20,38 @@ describe('API Service', function() {
     $httpBackend.verifyNoOutstandingRequest();
   });
 
+  describe('ensure api requests generate normalized urls', function() {
+    let expected: ng.IRequestConfig;
+    beforeEach(() => {
+      expected = {
+        method: '',
+        url: '',
+      };
+    });
+
+    it('trims leading slashes from urls', function() {
+      const result = API.one('/foo');
+      expected.url = `${baseUrl}/foo`;
+      expect(result.config).toEqual(expected);
+    });
+
+    it('trims repeated leading slashes from urls', function() {
+      const result = API.one('/////foo');
+      expected.url = `${baseUrl}/foo`;
+      expect(result.config).toEqual(expected);
+    });
+
+    it('trims trailing slashes from baseUrl', function() {
+      const originalUrl = SETTINGS.gateUrl;
+      SETTINGS.gateUrl = 'http://localhost/';
+      const result = API.one('foo');
+      SETTINGS.gateUrl = originalUrl;
```

Comment:
feels like there is a better way to do this, but this test would side-effect all the other tests unless I cleaned up this change to SETTINGS.gateUrl but I don't really understand the mocking/testing framework well enough to figure out how to customize the top level beforeEach that is setting the config to the value of SETTINGS.gateUrl just for this one test....

Reply:
Slightly better way: in the `afterEach` method of this test, call `SETTINGS.resetToOriginal();`

Reply:
I added that, though I also change the mock's baseUrl to `API.baseUrl` so this becomes unnecessary in this test for it to continue to pass (but feels like still the right thing to do if I'm mucking with SETTINGS)